arduino-hal/Cargo.toml

```diff
@@ -12,6 +12,7 @@ board-selected = []
 mcu-atmega=[]
 arduino-uno = ["mcu-atmega", "atmega-hal/atmega328p", "board-selected"]
 arduino-leonardo = ["mcu-atmega", "atmega-hal/atmega32u4", "board-selected"]
+arduino-mega = ["mcu-atmega", "atmega-hal/atmega2560", "board-selected"]
```

Comment:
The official name is _"Arduino Mega 2560"_ so I think we should reflect this by calling the feature-flag and assorted things `arduino-mega2560`... Just to prevent confusion in the future, should a different version of a "Mega" board ever be released.

Reply:
Renamed